### PR TITLE
Hackaton100219

### LINKED ISSUE #13: Ignore empty space after a name
For example:
in the measurement data file I have the name 'IR1_P ' and in the SBML model I have 'IR1_P'. It would be good to be able to automatically ignore spaces at the end of the string.

- to avoid:
Unknown observables in measurement file: {'observable_IR1_P '}

### PR DESCRIPTION
**Pull request by elbaraim**
New features:
- Checking presence of empty spaces at end of column names and column entry names - close #13 
- Check if all files exist when running `petablint.py`. Only if all files exist, the data format checking will be executed. TO DO: execute individual file checking.

### CODE REVIEW

File: petab/lint.py [hunk 1]
```diff
@@ -14,6 +15,17 @@ def _check_df(df, req_cols, name):
         raise AssertionError(
             f"Dataframe {name} requires the columns {missing_cols}.")
 
+def assert_presence_of_empty_spaces(names_list, name):
```
Comment (dweindl) [inline petab/lint.py]:
```suggestion
def assert_no_trailing_whitespace(names_list, name):
```

File: petab/lint.py [hunk 2]
```diff
@@ -14,6 +15,17 @@ def _check_df(df, req_cols, name):
...
+    r = re.compile("\w+\s$")
+    names_with_empty_string = list(filter(r.match, names_list))
+
+    if len(names_with_empty_string) > 0:
+        if name in ["condition", "parameter", "measurement"]:
+            raise AssertionError(
+                f"Empty spaces found at end of string in column names of {name} file.")
```
Comment (dweindl) [inline petab/lint.py]:
```suggestion
                f"Trailing whitespace in column names of {name} file.")
```

File: petab/lint.py [hunk 3]
```diff
@@ -24,13 +36,24 @@ def check_condition_df(df):
...
 
 def check_measurement_df(df):
     req_cols = [
         "observableId", "preequilibrationConditionId", "simulationConditionId",
         "measurement", "time", "observableParameters", "noiseParameters",
         "observableTransformation"
     ]
+
+    for column_name in req_cols:
+        if df[column_name].dtype != np.dtype(np.float) and df[column_name].dtype != np.dtype(np.int):
```
Comment (dweindl) [inline petab/lint.py]: Check for common numeric supertype using numpy or Numbers.

File: petab/lint.py [hunk 4]
```diff
@@ -46,10 +69,17 @@ def check_parameter_df(df):
             f"Parameter table has wrong index {df.index.name}."
             "expected 'parameterId'.")
 
+    assert_presence_of_empty_spaces(df.index.values, "parameterId")
```
Comment (dweindl) [inline petab/lint.py]: Isn't that already included in the following loop?

Comment (elbaraim) [reply petab/lint.py]: parameterId is not included in the `req_cols` list

File: petab/lint.py (hunk 1 above)
Comment (dweindl) [inline petab/lint.py]: Please add a test case for this function

File: petab/lint.py (hunk 3 above)
Comment (elbaraim) [reply petab/lint.py]: I used the following now:
`np.issubdtype(df[column_name].dtype, np.number)`

Comment (dweindl) [reply petab/lint.py]: Yeah, that's good

File: petab/lint.py (hunk 1 above)
Comment (elbaraim) [reply petab/lint.py]: I just added the test in `tests/test_lint.py`
could you have a look?

**Pull request merged**